packit/ogr_services.py

```diff
@@ -25,13 +35,74 @@ def decorator_check_readonly(class_object) -> Any:
     @functools.wraps(class_object)
     def output_class(*args, **kwargs):
         if kwargs[READ_ONLY_NAME]:
-            check_read_only_support()
+            check_read_only_support(class_object)
             return class_object(*args, **kwargs)
         kwargs.pop(READ_ONLY_NAME)
         return class_object(*args, **kwargs)
 
     return output_class
 
 
-GithubService: GithubServiceOrigin = decorator_check_readonly(GithubServiceOrigin)
 PagureService: PagureServiceOrigin = decorator_check_readonly(PagureServiceOrigin)
+
+
+class BetterGithubIntegration(github.GithubIntegration):
```

Comment:
Please, document this stuff a little. I would like to see the reason for this "magic"..;-)

Reply:
yeah, we should upstream that function at some point; the thing is that pygithub is dead to some extent :/